pkg/virtctl/guestfs/guestfs.go

```diff
@@ -182,29 +175,30 @@ func (c *guestfsCommand) run(cmd *cobra.Command, args []string) error {
 	if err != nil {
 		return err
 	}
-	if image == "" {
-		if err = ImageSetFunc(client.VirtClient); err != nil {
+	if c.image == "" {
+		c.image, err = ImageSetFunc(client.VirtClient)
+		if err != nil {
 			return err
 		}
 	}
-	fmt.Printf("Use image: %s \n", image)
-	exist, _ := client.existsPVC(pvc, namespace)
+	fmt.Printf("Use image: %s \n", c.image)
+	exist, _ := client.existsPVC(c.pvc, namespace)
 	if !exist {
-		return fmt.Errorf("The PVC %s doesn't exist", pvc)
+		return fmt.Errorf("The PVC %s doesn't exist", c.pvc)
 	}
-	inUse, err = client.isPVCinUse(pvc, namespace)
+	inUse, err = client.isPVCinUse(c.pvc, namespace)
 	if err != nil {
 		return err
 	}
 	if inUse {
-		return fmt.Errorf("PVC %s is used by another pod", pvc)
+		return fmt.Errorf("PVC %s is used by another pod", c.pvc)
 	}
-	isBlock, err := client.isPVCVolumeBlock(pvc, namespace)
+	isBlock, err := client.isPVCVolumeBlock(c.pvc, namespace)
 	if err != nil {
 		return err
 	}
-	defer client.removePod(namespace)
-	return client.createInteractivePodWithPVC(pvc, image, namespace, "/entrypoint.sh", []string{}, isBlock)
```

Comment:
In this case I'd stick to passing `pvc` and `image` instead of the full command.